kroxylicious-filters/kroxylicious-oauthbearer-validation/src/main/java/io/kroxylicious/proxy/filter/oauthbearer/OauthBearerValidationFilter.java

```diff
@@ -84,7 +83,7 @@ public OauthBearerValidationFilter(ScheduledExecutorService executorService, Sha
     public CompletionStage<RequestFilterResult> onSaslHandshakeRequest(short apiVersion, RequestHeaderData header, SaslHandshakeRequestData request,
                                                                        FilterContext context) {
         try {
-            if (OAUTHBEARER_MECHANISM.equals(request.mechanism())) {
+            if (OAUTHBEARER_MECHANISM.equals(request.mechanism()) && this.validateAuthentication) {
```

Comment:
if we are in the SASL reauth case, we don't want to create a SASL server.

Reply:
It occured to me also that this should verify that saslServer == null, if no, returning an ISE is appropriate as that would indicated a protocol violation.